### PR TITLE
Wrapped Client Side Request Statistics

### PR DESCRIPTION
**Pull request by kushagraThapar**
Wrapped Client Side Request Statistics in Cosmos Response Diagnostics. 
Exposed only toString() and requestLatency()

### CODE REVIEW

File: sdk/src/main/java/com/azure/data/cosmos/CosmosResponseDiagnosticStatistics.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    void clientSideRequestStatistics(ClientSideRequestStatistics clientSideRequestStatistics) {
+        this.clientSideRequestStatistics = clientSideRequestStatistics;
+    }
+
+    /**
+     * Retrieves Response Diagnostic String
+     * @return Response Diagnostic String
+     */
+    public String getCosmosResponseDiagnosticString() {
```
Comment (christopheranderson) [inline sdk/src/main/java/com/azure/data/cosmos/CosmosResponseDiagnosticStatistics.java]: let's keep it simple and make this just `toString`

File: sdk/src/main/java/com/azure/data/cosmos/CosmosResponseDiagnosticStatistics.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    CosmosResponseDiagnosticStatistics() {
+        this.clientSideRequestStatistics = new ClientSideRequestStatistics();
+    }
+
+    ClientSideRequestStatistics clientSideRequestStatistics() {
+        return clientSideRequestStatistics;
+    }
+
+    void clientSideRequestStatistics(ClientSideRequestStatistics clientSideRequestStatistics) {
```
Comment (christopheranderson) [inline sdk/src/main/java/com/azure/data/cosmos/CosmosResponseDiagnosticStatistics.java]: just for consistency's sake, this should `return this`

File: sdk/src/main/java/com/azure/data/cosmos/CosmosResponseDiagnosticStatistics.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+     */
+    public String getCosmosResponseDiagnosticString() {
+        return this.clientSideRequestStatistics.toString();
+    }
+
+    /**
+     * Retrieves latency related to the completion of the request
+     * @return request completion latency
+     */
+    public Duration getRequestLatency() {
```
Comment (christopheranderson) [inline sdk/src/main/java/com/azure/data/cosmos/CosmosResponseDiagnosticStatistics.java]: drop the `get`

File: sdk/src/main/java/com/azure/data/cosmos/CosmosResponse.java [hunk 4]
```diff
@@ -122,8 +122,8 @@ public String sessionToken(){
      *
      * @return diagnostics information for the current request to Azure Cosmos DB service.
      */
-    public String requestDiagnosticsString() {
-        return resourceResponseWrapper.getRequestDiagnosticsString();
+    public String cosmosResponseDiagnosticsString() {
```
Comment (christopheranderson) [inline sdk/src/main/java/com/azure/data/cosmos/CosmosResponse.java]: I was expecting us to expose the CosmosResponseDiagnostics here, not just a string.

File: sdk/src/main/java/com/azure/data/cosmos/CosmosResponseDiagnosticStatistics.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+ * SOFTWARE.
+ */
+package com.azure.data.cosmos;
+
+import java.time.Duration;
+
+/**
+ * This class represents response diagnostic statistics associated with a request to Azure Cosmos DB
+ */
+public class CosmosResponseDiagnosticStatistics {
```
Comment (christopheranderson) [inline sdk/src/main/java/com/azure/data/cosmos/CosmosResponseDiagnosticStatistics.java]: CosmosResponseDiagnostics

File: sdk/src/main/java/com/azure/data/cosmos/FeedResponseDiagnostics.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+    FeedResponseDiagnostics(Map<String, QueryMetrics> queryMetricsMap) {
+        this.queryMetricsMap = queryMetricsMap;
+    }
+
+    Map<String, QueryMetrics> queryMetricsMap() {
+        return queryMetricsMap;
+    }
+
+    public FeedResponseDiagnostics queryMetricsMap(Map<String, QueryMetrics> queryMetricsMap) {
```
Comment (christopheranderson) [inline sdk/src/main/java/com/azure/data/cosmos/FeedResponseDiagnostics.java]: should be package private?

Comment (kushagraThapar) [reply sdk/src/main/java/com/azure/data/cosmos/FeedResponseDiagnostics.java]: Yes, it should be.

**Pull request merged**